sdk/lib/conutils/pager.c

```diff
@@ -31,7 +33,127 @@
 // Temporary HACK
 #define CON_STREAM_WRITE    ConStreamWrite
 
+#define CP_SHIFTJIS 932  // Japanese Shift-JIS
+#define CP_HANGUL   949  // Korean Hangul/Wansung
+#define CP_JOHAB    1361 // Korean Johab
+#define CP_GB2312   936  // Chinese Simplified (GB2312)
+#define CP_BIG5     950  // Chinese Traditional (Big5)
 
+/* IsFarEastCP(CodePage) */
+#define IsCJKCodePage(CodePage) \
+    ((CodePage) == CP_SHIFTJIS || (CodePage) == CP_HANGUL || \
+  /* (CodePage) == CP_JOHAB || */ \
+     (CodePage) == CP_BIG5     || (CodePage) == CP_GB2312)
+
+static inline INT
+GetWidthOfCharCJK(
+    IN UINT nCodePage,
+    IN WCHAR ch)
+{
+    INT ret = WideCharToMultiByte(nCodePage, 0, &ch, 1, NULL, 0, NULL, NULL);
+    if (ret == 0)
+        ret = 1;
+    else if (ret > 2)
+        ret = 2;
+    return ret;
+}
+
+static BOOL __stdcall
+ConDefaultPagerLine(
+    IN OUT PCON_PAGER Pager,
+    IN PCTCH line,
+    IN DWORD cch,
+    IN OUT PDWORD pdwFlags)
```

Comment:
(This function and the other ones as well): Parameter list formatted as in the rest of the conutils library.

Reply:
Also, using "PCTCH" as in the rest of the lib: pointer to const "TCHAR" (note: there is no "PCTCHAR" type), as the buffer is NOT a null-terminated string (like what's understood with "PCTSTR" and the like), but a buffer of (T)chars.

Reply:
OK!